atmos_model.F90

```diff
@@ -0,0 +1,2743 @@
+                                         zorlmin = 1.0e-7_GFS_kind_phys
+
+contains
+
+!#######################################################################
+! <SUBROUTINE NAME="update_radiation_physics">
+!
+!<DESCRIPTION>
+!   Called every time step as the atmospheric driver to compute the
+!   atmospheric tendencies for dynamics, radiation, vertical diffusion of
+!   momentum, tracers, and heat/moisture.  For heat/moisture only the
+!   downward sweep of the tridiagonal elimination is performed, hence
+!   the name "_down".
+!</DESCRIPTION>
+
+!   <TEMPLATE>
+!     call  update_atmos_radiation_physics (Atmos)
+!   </TEMPLATE>
+
+! <INOUT NAME="Atmos" TYPE="type(atmos_data_type)">
+!   Derived-type variable that contains fields needed by the flux exchange module.
+!   These fields describe the atmospheric grid and are needed to
+!   compute/exchange fluxes with other component models.  All fields in this
+!   variable type are allocated for the global grid (without halo regions).
+! </INOUT>
+
+subroutine update_atmos_radiation_physics (Atmos)
+!-----------------------------------------------------------------------
+  type (atmos_data_type), intent(in) :: Atmos
+!--- local variables---
+    integer :: nb, jdat(8), rc, ierr
+
+    if (mpp_pe() == mpp_root_pe() .and. debug) write(6,*) "statein driver"
+!--- get atmospheric state from the dynamic core
+    call set_atmosphere_pelist()
+    call mpp_clock_begin(getClock)
+    if (GFS_control%do_skeb) call atmosphere_diss_est (GFS_control%skeb_npass) !  do smoothing for SKEB
+    call atmos_phys_driver_statein (GFS_data, Atm_block, flip_vc)
+    call mpp_clock_end(getClock)
+
+!--- if dycore only run, set up the dummy physics output state as the input state
+    if (dycore_only) then
+      do nb = 1,Atm_block%nblks
+        GFS_data(nb)%Stateout%gu0 = GFS_data(nb)%Statein%ugrs
+        GFS_data(nb)%Stateout%gv0 = GFS_data(nb)%Statein%vgrs
+        GFS_data(nb)%Stateout%gt0 = GFS_data(nb)%Statein%tgrs
+        GFS_data(nb)%Stateout%gq0 = GFS_data(nb)%Statein%qgrs
+      enddo
+    else
+      if (mpp_pe() == mpp_root_pe() .and. debug) write(6,*) "setup step"
+
+!--- update GFS_control%jdat(8)
+      jdat(:) = 0
+      call get_date (Atmos%Time, jdat(1), jdat(2), jdat(3),  &
+                                 jdat(5), jdat(6), jdat(7))
+      GFS_control%jdat(:) = jdat(:)
+
+!--- execute the atmospheric setup step
+      call mpp_clock_begin(setupClock)
+      call CCPP_step (step="timestep_init", nblks=Atm_block%nblks, ierr=ierr)
+      if (ierr/=0)  call mpp_error(FATAL, 'Call to CCPP timestep_init step failed')
+
+!--- call stochastic physics pattern generation / cellular automata
+      call stochastic_physics_wrapper(GFS_control, GFS_data, Atm_block, ierr)
+      if (ierr/=0)  call mpp_error(FATAL, 'Call to stochastic_physics_wrapper failed')
+
+!--- if coupled, assign coupled fields
+
+      if( GFS_control%cplflx .or. GFS_control%cplwav ) then
+
```

Comment:
Can we do this by default or have these booleans set to .true. when external fields are requested via NUOPC?